src/main/groovy/com/urbancode/air/plugin/cf/helper/CFHelper.groovy

```diff
@@ -776,9 +697,92 @@ class CFHelper {
         helper.runCommand("[Action] Running command: ${commandArgs.join(' ')}", commandArgs, setOauthToken)
         println "----------------------------------------------"
+        
+        
+        //Get organizations
+        try {
+            def organizationsUrl = new URL(api + '/v2/organizations')
+            HttpURLConnection connection = (HttpURLConnection) organizationsUrl.openConnection();
+            connection.addRequestProperty("Accept", "application/json")
+            connection.addRequestProperty("Authorization", oauthToken.trim())
+            connection.setRequestMethod('GET')
+            connection.setDoInput(true)
+            if(!connection.getResponseCode().equals(200)) {
+
+                println ("[Error] Bad response code of ${connection.getResponseCode()}.")
+                println ('Response:\n' + connection.getResponseMessage())
+                System.exit(1)
+
+            }
+
+            def organizationsJson = connection.getInputStream().getText().toString()
+
+            def organizationsSlurper = new JsonSlurper()
+            def organizations = organizationsSlurper.parseText(organizationsJson)
+
+            organizations.resources.each { org ->
+                if(org.entity.name.equals(organization)) {
+                    orgId = org.metadata.guid
+                }
+
+            }
+            
+        }
+        catch (IOException ex) {
+            println "[Error] I/O Error found when retrieving the organizations list. Please review the output log."
+            ex.printStackTrace()
+            System.exit(1)
+        }
+        catch (Exception ex) {
+            println "[Error] Unknown found when retrieving the organizations list. Please review the output log."
+            ex.printStackTrace()
+            System.exit(1)
+        }
+
+        //Get spaces
+        try {
+            def spacesUrl = new URL(api + '/v2/spaces')
+            HttpURLConnection connection = (HttpURLConnection) spacesUrl.openConnection();
+            connection.addRequestProperty("Accept", "application/json")
+            connection.addRequestProperty("Authorization", oauthToken.trim())
+            connection.setRequestMethod('GET')
+            connection.setDoInput(true)
+            if(!connection.getResponseCode().equals(200)) {
+
+                println ("[Error] Bad response code of ${connection.getResponseCode()}.")
+                println ('Response:\n' + connection.getResponseMessage())
+                System.exit(1)
+
+            }
+
+            def spacesJson = connection.getInputStream().getText().toString()
+
+            def spacesSlurper = new JsonSlurper()
+            def space = spacesSlurper.parseText(spacesJson)
+
+            space.resources.each { sp ->
+                if(sp.entity.organization_guid.equals(orgId)) {
+                    spaces << sp.metadata.guid
+                }
+
+            }
+            
+        }
```

Comment:
Why do we use the REST API to get organizations and spaces only when we acquire services? There's an existing function for getSpaces and getOrgs that uses the CLI. Are these functions no longer working? 

If so, you should replace the getSpaces and getOrgs logic with the logic you've written to get them with the REST API. (Assuming that this solution will work in all cases). Let me know if there is some reason that we need to keep the logic separated for different occasions.

Reply:
Old code is filtering out the required things (orgs, spaces, apps etc) from command line output. Due to change in output format of cloudfoundary commands, logic are failing to fetch the expected output. That's why I have made use of rest apis . Should I comment that old code and or replace with new one.

Reply:
Yes, if the command line logic is no longer working with the latest version of Cloud Foundry you should replace the old logic with the new logic. We don't want to have multiple methods to do the same thing, especially if the old method is no longer working.

Reply:
We were facing issue in getting services through command line, So I am fetching it through api, now api needs org id and space id to filter the services from service api, But methods getOrgs and getSpaces does not return id which is required to fetch services through api. They only return names. Leaving those two methods as it is and removing everything related to getServices, keeping the new method for fetching services and renaming it to "getServices", so it wont affect any other dependencies.